### ISSUE TITLE
(Twizy) Research Hardware replacement

### ISSUE THREAD

**Issue by davbt** [Monash-Connected-Autonomous-Vehicle/autoware#58]
I am not exactly sure of our budget (yet), but I'd aim for no more than $1500 for now. 

#### Alienware Laptop
- Motive: mediocre performance running the Camera Stream (20 fps on RViz). Imagine what would happen if we ran the entire Autoware program 
- [Autoware's  Minimum Hardware Requirements ](https://autowarefoundation.github.io/autoware-documentation/main/installation/#minimum-hardware-requirements) (16GB ram, 4 GB Nvidia GPU, 8-core CPU)
- [Recommended Rviz specs](https://answers.ros.org/question/370953/2021-recommended-specifications-for-a-pc-to-manage-a-telemanipulation-setup-in-ros-nodes-tf-rviz-and-rosbag/) (32GB ram) 
Ideally, it should not consume much power from Twizy's battery, and even better if it doesn't need to use any power at all (e.g., laptops).
- [ ] Research the costs of RAM and CPU upgrades on Alienware laptop
- [ ] General research for entire laptop replacement that satisfies the requirements above

#### Drive PX 2
- Motive: AFAIK, can only use ROS 1 while still having the Nvidia driveworks packages to run gmsl cameras. A replacement is not needed if our primary computer can run a streamer.
- GMSL-USB converters do exist but are [rare & expensive](https://www.amazon.co.jp/dp/B0D3KKLN2Y?tag=press20240626-22)
- [ ] Research potential PX 2 replacements that can support GMSL cameras 
 
#### Or, scale horizontally (keep our setup and add more hardware):
- e.g. one laptop for sensors management, px 2 for cameras, keep Alienware to run Autoware
- This is free since we already have unused laptops/desktops in the workshop
- [x] Get specs and post on this issue any unused laptops/PCs and their specs (david)

**Comment by davbt** [comment]
Hi all! Add your name to the tasks you want to do (edit the post above then add (`<your-name>`) beside any task `- [ ]`). This should be chill, we should aim to finish this research within the next 7 days and add any useful things to this GitHub issue.

**Comment by amrsnch** [comment]
Yo, I was wondering if we can go with a refurbished machine to save some money. Also, considering that it's going to be stored in the back of the Twizy and not used directly, it kind of makes sense. However, it also depends on the budget we have to spend, lol. If we can afford a new machine, we can go with that as well. A PC is also under consideration.

**Comment by amrsnch** [comment]
Also, I've got a recommendation regarding Raspberry alternatives. You can check them out here: https://www.khadas.com/vim4

**Comment by davbt** [comment]
> Yo, I was wondering if we can go with a refurbished machine to save some money. Also, considering that it's going to be stored in the back of the Twizy and not used directly, it kind of makes sense.

Yep, definitely. It could even have a broken screen since it would still be usable via Anydesk or ssh

> However, it also depends on the budget we have to spend, lol. If we can afford a new machine, we can go with that as well. A PC is also under consideration.

Pros of PC: cheap and easier to upgrade. Cons: constant battery consumption and size.

> Also, I've got a recommendation regarding Raspberry alternatives. You can check them out here: https://www.khadas.com/vim4

This seems pretty interesting. We could have one these to solely manage the lidar or cameras. I'll think about this for a bit. Good find!

**Comment by davbt** [comment]
The fyp teams have several Raspberry Pis
![IMG_0829](https://github.com/user-attachments/assets/bc8ace6b-0fef-404a-a35e-ac1c0631f7b8)
![IMG_0828](https://github.com/user-attachments/assets/54096935-aa06-4c52-bc3b-819d112ae327)

And there is a 16GB RAM laptop with Ubuntu, now in G03
![IMG_0830](https://github.com/user-attachments/assets/7ca33c7f-82ed-4b0d-9d40-9a267d14240d)

**Comment by davbt** [comment]
Fixed issue on why we couldn't ping the PX 2 from the new Dell Latitude laptop:
- [IP address wasn't in the same subnet](https://superuser.com/a/375165) as the px2. (Latitude: **192.168.10**.100, PX 2: **192.168.1**.181)
- Fixed with `sudo ip addr add 192.168.1.100/24 dev enp0s31f6`, i.e. changing ip address of interface to have same subnet as px 2 ([source](https://askubuntu.com/a/1243748))

I'll leave it to one of you guys to try running sensors from this laptop. I'll hold off from buying a new computer.

Edit: I'll set up anydesk now and keep the laptop on if you wanna install some things. Also VPN is set up. I add the credentials on [this page](https://www.notion.so/monashcav/Twizy-3dc0bf8de24c4e38acb5b78ab59edda5).
